dbms/src/DataStreams/ParallelAggregatingBlockInputStream.cpp

```diff
@@ -235,4 +235,14 @@ void ParallelAggregatingBlockInputStream::appendInfo(FmtBuffer & buffer) const
     buffer.fmtAppend(", max_threads: {}, final: {}", max_threads, final ? "true" : "false");
 }
 
+uint64_t ParallelAggregatingBlockInputStream::collectCPUTimeNsImpl(bool is_root)
+{
+    uint64_t cpu_time_ns = impl ? impl->collectCPUTimeNs(is_root) : 0;
+    forEachChild([&](IBlockInputStream & child) {
+        cpu_time_ns += child.collectCPUTimeNs(true);
```

Comment:
So the cpu time is defined as the sum of execution time of all threads?

Reply:
Yes, comments added.
```
    /** Estimate the cpu time nanoseconds used by block input stream dag.
      * In this method, streams are divided into two categories:
      * - thread-runner: Called directly by a thread
      * - non-thread-runner: Called by a thread-runner
      * Here we should count the execution time of each thread-runner.
      * Note: Because more threads than vcore, and blocking relationships between streams,
      * the result may not be 100% identical to the actual cpu time nanoseconds.
      */
```